src/msg/async/Stack.cc

```diff
@@ -31,13 +31,12 @@
 #undef dout_prefix
 #define dout_prefix *_dout << "stack "
 
-void NetworkStack::add_thread(unsigned i, std::function<void ()> &thread)
+std::function<void ()> NetworkStack::add_thread(unsigned i)
 {
   Worker *w = workers[i];
-  thread = std::move(
-    [this, w]() {
-      const uint64_t EventMaxWaitUs = 30000000;
-      w->center.set_owner();
+  std::function<void ()> thread = [this, w]() {
```

Comment:
One could, it wouldn't hurt anything, this seemed clearer. If we were using C++14 we could just set the return type to 'auto' and return the lambda without packing it into a std::function for the caller to do what they like with.